bot/exts/moderation/incidents.py

```diff
@@ -146,6 +208,35 @@ async def add_signals(incident: discord.Message) -> None:
             await incident.add_reaction(signal_emoji.value)
 
 
+async def extract_message_links(message: discord.Message, bot: Bot) -> t.Optional[list]:
+    """
+    Checks if there's any message links in the text content.
+
+    Then passes the the message_link into `make_message_link_embed` to format a
+    embed for it containing information about the link.
+
+    As discord only allows a max of 10 embeds in a single webhook, just send the
+    first 10 embeds and don't care about the rest.
+
+    If no links are found for the message, it logs a trace statement.
+    """
+    message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
+
+    if message_links:
+        embeds = []
+        for message_link in message_links:
+            ctx = await bot.get_context(message)
+            embed = await make_message_link_embed(ctx, message_link[0])
+            if embed:
+                embeds.append(embed)
+
+        return embeds[:10]
```

Comment:
Instead of processing all and then returning the first 10, it may make more sense to return after processing 10.

Reply:
Resolved in 248ecbf.